src/server/auth.ts

```diff
@@ -53,3 +54,42 @@ export default async function initializePatronTokenAuth(req: NextRequest) {
 
   return patronTokenResponse
 }
+
+/**
+ * Creates and returns redirect urk. Call this function only with an un-authenticated patron,
+ * i.e., patronTokenResponse.isTokenValid must be false.
+ */
+export function getLoginRedirect(req) {
+  const protocol = req.protocol || "http"
+  const hostname = req.headers["host"]
+  const originalUrl = BASE_URL + req.url
+  const fullUrl = encodeURIComponent(`${protocol}://${hostname}${originalUrl}`)
+  const redirect = `${
+    appConfig.externalUrls.loginUrl[appConfig.environment]
+  }?redirect_uri=${fullUrl}`
+  return redirect
+}
+
+/**
+ * Creates redirect to log out user, then return user to their current page.
+ */
+export const useLogoutRedirect = () => {
+  // Will send user back to prod if below doesn't work
```

Comment:
Say that typically the below, the useEffect, won't work for the noscript users (or the users who have javascript turned off).